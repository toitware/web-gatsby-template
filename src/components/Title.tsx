import { makeStyles, Typography } from "@material-ui/core";
import { ReactNode } from "react";
import * as React from "react";

// This style is just added for reference.
const useStyles = makeStyles((theme) => ({
  title: {
    borderBottom: `2px solid ${theme.palette.text.primary}`,
  },
}));

type Props = {
  children: ReactNode;
};

export function Title({ children }: Props): JSX.Element {
  const classes = useStyles();
  return (
    <Typography className={classes.title} variant="h1">
      {children}
    </Typography>
  );
}

export default Title;
