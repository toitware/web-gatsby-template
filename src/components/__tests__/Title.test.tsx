import { render, screen } from "@testing-library/react";
import React from "react";
import Title from "../Title";

describe("Title", () => {
  it("renders correctly", () => {
    const result = render(<Title>title content</Title>);

    expect(result.container.firstChild).toMatchSnapshot();

    expect(screen.getByText("title content")).toBeInstanceOf(HTMLHeadingElement);
  });
});
