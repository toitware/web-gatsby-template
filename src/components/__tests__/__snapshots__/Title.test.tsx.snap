// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<h1
  class="MuiTypography-root makeStyles-title MuiTypography-h1"
>
  title content
</h1>
`;
